repos/system_upgrade/common/actors/checkconsumedassets/libraries/check_consumed_assets.py

```diff
@@ -0,0 +1,128 @@
+from collections import defaultdict, namedtuple
+
+from leapp import reporting
+from leapp.libraries.common.config import get_consumed_data_stream_id
+from leapp.libraries.common.fetch import ASSET_PROVIDED_DATA_STREAMS_FIELD
+from leapp.libraries.stdlib import api
+from leapp.models import ConsumedDataAsset
+
+
+def report_incompatible_asset_of_same_type(incompatibility_characterization, assets, are_assets_malformed=False):
+    if not assets:
+        return
+
+    title = 'The asset{multiple_suffix} {which_assets} {verb} {reason}'
```

Comment:
title should not be affected singular & plural forms. The difference between one asset or several assets are e.g. incompatible, is irrelevant. The issue is same, it has only a different parameters. Imagine having a script that is detecting whether this issue has been hit or not. User is supposed to detect the particular problem in the report checking the `key` value. Which is also affected by the `title`, because the title is supposed to be invariant as  the problem. Keeping the review for the next week, but the function seems to me too much generic just from a short look.

Reply:
Thanks for the detailed explanation on static report titles, I've gained another piece of knowledge and perspective.

I've crystalized the generic function into two separate functions, hoping to make reporting less convoluted.

Reply:
@MichalHe it seems really good now! thanks :)